client/app/components/ParameterValueInput.jsx

```diff
@@ -39,16 +39,19 @@ export class ParameterValueInput extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      value: props.value,
-      isDirty: false,
+      value: props.parameter.pendingValue || props.value,
+      isDirty: props.parameter.hasPendingValue,
     };
   }
 
   componentDidUpdate = (prevProps) => {
-    const { value } = this.props;
+    const { value, parameter } = this.props;
     // if value prop updated, reset dirty state
-    if (prevProps.value !== value) {
-      this.setState({ value, isDirty: false });
+    if (prevProps.value !== value || prevProps.parameter !== parameter) {
```

Comment:
I think that now there's no actual need for any condition, right?

Reply:
It is also called when state is update, removing it would cause an infinite loop :slightly_smiling_face: